test/Microsoft.Extensions.ProjectModel.Tests/MsBuild/MsBuildProjectDependencyProviderTests.cs

```diff
@@ -0,0 +1,171 @@
+                fileProvider.Add($"Root{Path.DirectorySeparatorChar}Excluded.cs", "public class Abc {}");
+
+                // Add Class Library project
+                fileProvider.Add($"Library1{Path.DirectorySeparatorChar}library.csproj", LibraryProjectTxt);
+                fileProvider.Add($"Library1{Path.DirectorySeparatorChar}Three.cs", "public class Abc3 {}");
+
+                var testContext = _fixture.GetMsBuildContext();
+
+                var muxer = Path.Combine(testContext.ExtensionsPath, "../..", "dotnet.exe");
+                var result = Command
+                    .Create(muxer, new[] { "restore3", Path.Combine(fileProvider.Root, "Library1","Library1.csproj") })
+                    .OnErrorLine(l => _output.WriteLine(l))
+                    .OnOutputLine(l => _output.WriteLine(l))
+                    .Execute();
+
+                Assert.Equal(0, result.ExitCode);
+
+                result = Command
+                    .Create(muxer, new[] { "restore3", Path.Combine(fileProvider.Root, "Root", "test.csproj") })
+                    .OnErrorLine(l => _output.WriteLine(l))
+                    .OnOutputLine(l => _output.WriteLine(l))
+                    .Execute();
+
+                Assert.Equal(0, result.ExitCode);
+
+                var builder = new MsBuildProjectContextBuilder()
+                    .AsDesignTimeBuild()
+                    .UseMsBuild(testContext)
+                    .WithTargetFramework(FrameworkConstants.CommonFrameworks.NetCoreApp10)
+                    .WithConfiguration("Debug")
+                    .WithProjectFile(fileProvider.GetFileInfo(Path.Combine("Root", "test.csproj")));
+
+                var context = builder.Build();
+
+                var compilationAssemblies = context.CompilationAssemblies;
+                var lib1Dll = compilationAssemblies
+                    .Where(assembly => assembly.Name.Equals("Library1", StringComparison.OrdinalIgnoreCase))
+                    .FirstOrDefault();
+
+                Assert.NotNull(lib1Dll);
+                var expectedPath = Path.Combine(fileProvider.Root, "Root", "bin", "Debug", "netcoreapp1.0", "lib1.dll");
+                Assert.Equal(expectedPath, lib1Dll.ResolvedPath, ignoreCase: true);
+                
+                // This reference doesn't resolve so should not be available here.
+                Assert.False(compilationAssemblies.Any(assembly => assembly.Name.Equals("xyz", StringComparison.OrdinalIgnoreCase)));
+
+                var packageDependencies = context.PackageDependencies;
+                var mvcPackage = packageDependencies
+                    .Where(p => p.Name.Equals("Microsoft.AspNetCore.Mvc", StringComparison.OrdinalIgnoreCase))
+                    .FirstOrDefault();
+                Assert.NotNull(mvcPackage);
+
+                Assert.True(mvcPackage.Dependencies.Any(dependency => dependency.Name.Equals("Microsoft.Extensions.DependencyInjection", StringComparison.OrdinalIgnoreCase)));
```

Comment:
Has the same effect, but you could also use
 `Assert.Contains(mvcPackage.Dependencies, d => d.Name.Equals("Microsoft.Extensions.DependencyInjection"))`